optimism/src/trace.rs

```diff
@@ -85,46 +86,50 @@ impl<
     }
 }
 
-pub(crate) trait Folder<const N: usize, Selector: Eq + Hash + Indexer + Copy, F: FftField> {
+pub(crate) trait Folder<const N: usize, C: FoldingConfig, Sponge, F: FftField> {
     /// Returns the witness for the given selector as a folding witness and
     /// folding instance pair.
     /// Note that this function will also absorb all commitments to the columns
     /// to coin challenges appropriately.
     fn to_folding_pair(
         &self,
-        selector: Selector,
-        srs: &SRS<Curve>,
-        fq_sponge: &mut BaseSponge,
-    ) -> (FoldingInstance<N>, FoldingWitness<N>);
+        selector: C::Selector,
+        srs: &poly_commitment::srs::SRS<C::Curve>,
+        sponge: &mut Sponge,
+    ) -> (
+        FoldingInstance<N, C::Curve>,
+        FoldingWitness<N, <<C as FoldingConfig>::Curve as AffineCurve>::ScalarField>,
+    );
 }
 
-impl<
-        const N: usize,
-        const N_REL: usize,
-        const N_SEL: usize,
-        Selector: Eq + Hash + Indexer + Copy + Ord + PartialOrd,
-    > Folder<N, Selector, Fp> for Trace<N, N_REL, N_SEL, Selector, Fp>
+impl<const N: usize, const N_REL: usize, const N_SEL: usize, C: FoldingConfig, Sponge>
+    Folder<N, C, Sponge, ScalarField<C>> for Trace<N, N_REL, N_SEL, C::Selector, ScalarField<C>>
+where
+    C::Selector: Ord + PartialOrd + Indexer,
```

Comment:
Wondering if `Ord + PartialOrd` should not be moved by default into folding. It is used by decomposable folding.

Reply:
yeah, I was wondering the same in the next PR I was working on... Will probably try that out and see if it all works fine.